File: sound-parent/sound-web/src/main/resources/static/js/service/roleService.js
//服务层
app.service('roleService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('/role/findAll');
	}
	//读取角色数据绑定到多选框中
	this.findForSelect2=function(){
		return $http.get('/role/findForSelect2');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('/role/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('/role/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('/role/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('/role/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('/role/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('/role/search?page='+page+"&rows="+rows, searchEntity);
	}
	//保存关联组信息
	this.saveGroup=function(roleid,ids){
		var entitynew={
			roleid:roleid,
			ids:ids
		}
		return $http.post('/rolegroup/addGroup',entitynew);
	}
	//保存关联资源信息
	this.savePermission=function(roleid,ids){
		var entitynew={
			roleid:roleid,
			ids:ids
		}
		return $http.post('/rolepermission/addPermission',entitynew);
	}
	//查询该roleid下的关联组信息
	this.findGroupOfRole=function (roleid) {
		return $http.get('/rolegroup/findGroupOfRole?id='+roleid)
	}
	//查询该roleid下的关联资源信息
	this.findPermissionOfRole=function (roleid) {
		return $http.get('/rolepermission/findPermissionOfRole?id='+roleid)
	}
});

File: sound-parent/sound-api/src/main/java/com/wisdom/sound/service/RoleService.java
package com.wisdom.sound.service;
import java.util.HashMap;
import java.util.List;
import java.util.Map;


import com.wisdom.sound.entity.PageResult;
import com.wisdom.sound.pojo.Role;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface RoleService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Role> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Role role);
	
	
	/**
	 * 修改
	 */
	public void update(Role role);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Role findOne(String id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(String[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Role role, int pageNum, int pageSize);
	/**
	 * 功能描述 : 返回所有角色
	 * 参数 :[] :
	 * @return java.util.List<com.wisdom.sound.pojo.Role>
	 * @date  2022/3/15
	 * @author  liuwisdom
	 */
    List<HashMap<String, String>> findForSelect2();

    void addRoleGroup(Map<String, Object> entitynew);

	void deleteRoleGroupByUserId(String roleId);

	List<Map> findGroupOfRole(String id);
}
